packages/jest/tsconfig.json

```diff
@@ -1,18 +1,3 @@
 {
-  "compilerOptions": {
-    "noEmit": true,
-    "declaration": true,
-    "sourceMap": false,
-    "alwaysStrict": true,
-    "noImplicitAny": true,
-    "noImplicitThis": true,
-    "removeComments": true,
-    "preserveConstEnums": true,
-    "skipLibCheck": true,
-    "strictNullChecks": true,
-    "downlevelIteration": true,
-    "module": "commonjs",
-    "target": "es2017",
-    "outDir": "lib/"
-  }
+  "extends": "../../tsconfig.shared.json"
```

Comment:
```diff
+++ "declaration": false,
+++ "incremental": false,
+++ "importHelpers": false,
+++ "noFallthroughCasesInSwitch": true,
+++ "noUnusedLocals": true,
+++ "strict": true,
--- "declaration": true,
--- "alwaysStrict": true,
--- "noImplicitAny": true,
--- "noImplicitThis": true,
--- "removeComments": true,
--- "downlevelIteration": true,
--- "outDir": "lib/",
```

Reply:
Can be summarized into:

As `strict` is equivalent to `alwaysStrict`, `strictNullChecks`, `strictBindCallApply`, `strictFunctionTypes`, `strictPropertyInitialization`, `noImplicitAny`, `noImplicitThis` and `useUnknownInCatchVariables`.

Reply:
- declaration: 👍 `tsconfig.publish.json` adapted accordingly
- incremental: no-op
- importHelpers: no impact
- noFallthroughCasesInSwitch: stricter compilation
- noUnusedLocals: stricter compilation
- strict: stricter compilation
- downlevelIteration: unused as target was es2017>es5
- removeComments: we want to preserve jsdoc, so putting it back to false should be a right change -> ⚠️ impact on the generated package